### PR TITLE
Enable security profiles in ARM template

### PR DESCRIPTION
**Pull request by harz566**
- Enable TVM deployment, and refine definition for 'TVM-TEST-SECUREBOOT-COMPATIBILITY' and 'VALIDATE-INTEL-SGX-DRIVER-FOR-DC-VM'
- Fix a few bugs for definition

=====================Test Logs ===========================

[LISAv2 Test Results Summary]
Test Run On           : 08/27/2020 23:46:42
ARM Image Under Test  : Canonical : UbuntuServer : 18_04-lts-gen2 : latest
Total Test Cases      : 3 (3 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:13

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 CORE                 VERIFY-DEPLOYMENT-PROVISION                                                       PASS                 2.11 
      ARMImageName: Canonical UbuntuServer 18_04-lts-gen2 latest, TestLocation: westus2, Kernel Version: 5.3.0-1035-azure
      FirstBoot : PASS
      FirstBoot : Call Trace Verification : PASS
      Reboot : PASS
      Reboot : Call Trace Verification : PASS
      
    2 TVM                  TVM-TEST-SECUREBOOT-COMPATIBILITY                                                 PASS                 0.79 
      ARMImageName: Canonical UbuntuServer 18_04-lts-gen2 latest, OverrideVMSize: Standard_D2s_v3, TestLocation: westus2, VMGeneration: 2, Kernel Version: 5.3.0-1035-azure
    3 SGX                  VALIDATE-INTEL-SGX-DRIVER-FOR-DC-VM                                               PASS                 4.15 
      ARMImageName: Canonical UbuntuServer 18_04-lts-gen2 latest, OverrideVMSize: Standard_DC2s_v2, TestLocation: uksouth, VMGeneration: 2, Kernel Version: 5.3.0-1035-azure

Logs can be found at C:\LISAv2OL\TestResults\2020-08-27-16-45-56-6662

08/28/2020 00:00:26 : [DEBUG] Creating 'C:\LISAv2OL\Azure-GM17-TestLogs.zip' from 'C:\LISAv2OL\TestResults\2020-08-27-16-45-56-6662'
08/28/2020 00:00:26 : [DEBUG] C:\LISAv2OL\Azure-GM17-TestLogs.zip created successfully.
08/28/2020 00:00:26 : [INFO ] Analyzing test results ...
08/28/2020 00:00:26 : [INFO ] LISAv2 exit code: 0

### CODE REVIEW

File: XML/TestCases/FunctionalTests-SGX.xml [hunk 1]
```diff
@@ -12,9 +12,9 @@
     <SetupConfig>
       <TestLocation>uksouth</TestLocation>
       <SetupType>OneVM</SetupType>
-      <OverrideVMSize>Standard_DC2s_v2</OverrideVMSize>
+      <OverrideVMSize>Standard_DC2s_v2,=~Standard_DC[0-9]+s_v2</OverrideVMSize>
       <VMGeneration>2</VMGeneration>
-      <ARMImageName>Canonical UbuntuServer 18_04-lts-gen2 latest</ARMImageName>
+      <ARMImageName>=~UbuntuServer.*gen2</ARMImageName>
```
Comment (juhlee-microsoft) [inline XML/TestCases/FunctionalTests-SGX.xml:17]: If you introduce the new name convention of OverrideVMSize and ARMImageName, please add the new context in README or howtodo document.

File: Libraries/Azure.psm1 [hunk 2]
```diff
@@ -1382,6 +1392,27 @@ Function Invoke-AllResourceGroupDeployments($SetupTypeData, $CurrentTestData, $R
 			Add-Content -Value "$($indents[4])}," -Path $jsonFile
 			#endregion
 
+			#region Security Profile
+			if ($CurrentTestData.SetupConfig.SecureBoot -or $CurrentTestData.SetupConfig.vTPM) {
+				Add-Content -Value "$($indents[4])^securityProfile^: " -Path $jsonFile
+				Add-Content -Value "$($indents[4]){" -Path $jsonFile
+				Add-Content -Value "$($indents[5])^uefiSettings^: " -Path $jsonFile
+				Add-Content -Value "$($indents[5]){" -Path $jsonFile
+				if ($CurrentTestData.SetupConfig.SecureBoot -and !$CurrentTestData.SetupConfig.vTPM) {
```
Comment (juhlee-microsoft) [inline Libraries/Azure.psm1:1401]: Should we check if both SecureBoot and vTPM are true condition? This is not correct configuration but we should check this faulty case in if-statement.

**Review by juhlee-microsoft — CHANGES_REQUESTED** [review]

File: Libraries/Azure.psm1 (hunk 2 above)
Comment (harz566) [reply Libraries/Azure.psm1:1401]: No, this is by design.  it's correct configuration even use only SecureBoot or vTPM.

File: XML/TestCases/FunctionalTests-SGX.xml (hunk 1 above)
Comment (harz566) [reply XML/TestCases/FunctionalTests-SGX.xml:17]: This is not newly introduced in this PR, but we should include more detailed instruction in ReadMe doc.  I have append changes in README, please take a look.

**Review by juhlee-microsoft — APPROVED** [review]

**Review by abhimarathe — APPROVED** [review]

**Review by lubaihua33 — APPROVED** [review]

**Pull request merged**